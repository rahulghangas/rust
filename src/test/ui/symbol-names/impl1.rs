#![feature(rustc_attrs)]
#![allow(dead_code)]

mod foo {
    pub struct Foo { x: u32 }

    impl Foo {
        #[rustc_symbol_name]
        //~^ ERROR symbol-name(_RNpMNdCs4fqI2P2rA04_5impl13fooNnB2_3Foo3bar)
        #[rustc_def_path] //~ ERROR def-path(foo::Foo::bar)
        fn bar() { }
    }
}

mod bar {
    use foo::Foo;

    impl Foo {
        #[rustc_symbol_name]
        //~^ ERROR symbol-name(_RNxMNoCs4fqI2P2rA04_5impl13barNnNdB4_3foo3Foo3baz)
        #[rustc_def_path] //~ ERROR def-path(bar::<impl foo::Foo>::baz)
        fn baz() { }
    }
}

fn main() {
}
