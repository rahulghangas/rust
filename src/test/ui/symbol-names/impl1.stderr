error: symbol-name(_RNpMNdCs4fqI2P2rA04_5impl13fooNnB2_3Foo3bar)
  --> $DIR/impl1.rs:8:9
   |
LL |         #[rustc_symbol_name]
   |         ^^^^^^^^^^^^^^^^^^^^

error: def-path(foo::Foo::bar)
  --> $DIR/impl1.rs:10:9
   |
LL |         #[rustc_def_path] //~ ERROR def-path(foo::Foo::bar)
   |         ^^^^^^^^^^^^^^^^^

error: symbol-name(_RNxMNoCs4fqI2P2rA04_5impl13barNnNdB4_3foo3Foo3baz)
  --> $DIR/impl1.rs:19:9
   |
LL |         #[rustc_symbol_name]
   |         ^^^^^^^^^^^^^^^^^^^^

error: def-path(bar::<impl foo::Foo>::baz)
  --> $DIR/impl1.rs:21:9
   |
LL |         #[rustc_def_path] //~ ERROR def-path(bar::<impl foo::Foo>::baz)
   |         ^^^^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

